x11-packages/xfce4-cpugraph-plugin/0001-cpu-data-thread-stat.patch

```diff
@@ -0,0 +1,181 @@
+diff --git a/panel-plugin/os.cc b/panel-plugin/os.cc
+index a8c9e40..3016be9 100644
+--- a/panel-plugin/os.cc
++++ b/panel-plugin/os.cc
+@@ -79,9 +79,176 @@ static kstat_ctl_t *kc;
+ #endif
+ 
+ #if defined (__linux__) || defined (__FreeBSD_kernel__)
++
++#ifdef __ANDROID__
++
++#include <sys/system_properties.h>
++#include <unistd.h>
++#include <glob.h>
++
++static int _sdkver = -1;
++
++void get_sdk_ver() {
++    char sdk_ver_str[PROP_VALUE_MAX];
++    __system_property_get("ro.build.version.sdk", sdk_ver_str);
++	_sdkver = atoi(sdk_ver_str);
++}
++
++static guint
++detect_cpu_number_android28 ()
++{
++    return sysconf(_SC_NPROCESSORS_ONLN);
++}
++
++static bool
++read_cpu_data_android28 (unordered_map<guint, CpuData> &data)
++{
++	static gulong cpu_core_time = 0, cpu_core_time_old = 0;
++    gulong cpu_core_time_delta = 0;
++	struct timespec ts;
++    glob_t globbuf;
++    int i;
++    int tid, processor;
++    gulong utime, stime;
++	static GHashTable *hash_cpu_user = NULL;
++	static GHashTable *hash_cpu_system = NULL;
++	gulong utime_old, stime_old;
++    FILE *fp;
++
++    if (G_UNLIKELY(data.size() == 0))
++        return false;
++
++    const size_t nb_cpu = data.size()-1;
++
++    for (i = 0; i <= nb_cpu; i++) {
++        data[i].user = 0;
++        data[i].system = 0;
++        data[i].load = 0;
++    }
++
++    // stocktake CPU time
++	gulong hertz = sysconf(_SC_CLK_TCK);
++	if (hertz <= 0) {
++#ifdef HZ
```

Comment:
Where would `HZ` have been defined?

Reply:
That's copied from `procps` (see below). `HZ` is probably not so meaningful for Android/Termux cuz that's probably for those who build a customized kernel. Also I should use `glong` not `gulong` otherwise the subsequent negativity test won't work. Will remove `HZ` and correct later. 

```cpp
/*
 * procps_hertz_get:
 *
 *
 * Some values in /proc are expressed in units of 1/HZ seconds, where HZ
 * is the kernel clock tick rate. One of these units is called a jiffy.
 * The HZ value used in the kernel may vary according to hacker desire.
 *
 * On some architectures, the kernel provides an ELF note to indicate
 * HZ.
 *
 * Returns:
 *  The discovered or assumed hertz value
 */
PROCPS_EXPORT long procps_hertz_get(void)
{
    long hz;

#ifdef _SC_CLK_TCK
    if ((hz = sysconf(_SC_CLK_TCK)) > 0)
        return hz;
#endif
#ifdef HZ
    return(HZ);
#endif
    /* Last resort, assume 100 */
    return 100;
}
```